zerver/lib/home.py

```diff
@@ -35,6 +38,23 @@ class UserPermissionInfo:
     show_webathena: bool
 
 
+def is_outdated_server(user_profile: Optional[UserProfile]) -> bool:
+    # TODO: We should ideally be using the minimum of this calculation
+    # and the date the release tarball was generated.
+    tzaware_last_upgrade_time = settings.LAST_SERVER_UPGRADE_TIME.replace(tzinfo=pytz.utc)
+    deadline = tzaware_last_upgrade_time + datetime.timedelta(
+        days=settings.SERVER_UPGRADE_NAG_DEADLINE
+    )
+
+    if user_profile is not None and user_profile.is_realm_admin:
+        # Administrators get warned a week early.
+        deadline = deadline - datetime.timedelta(days=7)
```

Comment:
Maybe make this 30 days?

Reply:
Yeah; also going to change it to add for non-admins rather than subtract.